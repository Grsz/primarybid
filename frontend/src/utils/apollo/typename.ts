export const removeTypenames = <T>(data: T): T =>
  JSON.parse(JSON.stringify(data), (key, value) =>
    key === '__typename' ? undefined : value,
  );
