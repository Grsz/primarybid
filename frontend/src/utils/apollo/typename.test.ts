import { removeTypenames } from './typename';

describe('typename', () => {
  describe('removeTypenames', () => {
    test('removes deeply nested "__typename" fields from graphql data', () => {
      const test = {
        __typename: 'foo',
        value: 'foo',
        bar: {
          __typename: 'bar',
          value: 'bar',
          baz: [
            {
              value: 'baz',
              __typename: 'baz',
            },
          ],
        },
      };

      expect(removeTypenames(test)).toMatchSnapshot();
    });
  });
});
