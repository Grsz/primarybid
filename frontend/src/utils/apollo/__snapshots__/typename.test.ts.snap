// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typename removeTypenames removes deeply nested "__typename" fields from graphql data 1`] = `
Object {
  "bar": Object {
    "baz": Array [
      Object {
        "value": "baz",
      },
    ],
    "value": "bar",
  },
  "value": "foo",
}
`;
